### DISCUSSION TITLE
"Native module cannot be null" exception is thrown when I install "react-native-tcp-socket" package

### DISCUSSION THREAD

**Discussion by arturmusin** [npm/feedback#440] [General]
Hi, I've been struggling with this issue over a week. I create empty react-native application for MacOS using this tutorial https://microsoft.github.io/react-native-windows/docs/rnm-getting-started

 and then install "react-native-tcp-socket" package.
When I try to use this library in my project I get 
**"Error: Requiring module "node_modules/react-native-tcp-socket/src/index.js", which threw an exception: 
Invariant Violation: Native module cannot be null."**

Could anyone please help me with that? I'm bashing my head against this wall over a week already...

`
import React, { Component } from 'react';
import {Text,View} from 'react-native';

const tcpSocket = require('react-native-tcp-socket');

class App extends Component {
  render() {
    return (
      <View>
        <Text>Hello React Native Sockets!</Text>
      </View>
    );
  }
}

export default App;
`

I do understand "react-native-tcp-socket" package is a wrapper around "net" module. So I also tried to install this module using "npm i net" command. But when I try to use any function from this "net" objects -- I see in console log that all functions are "undefined".... I'm completely lost in this ....

**Comment by MylesBorins** [comment]
Hey @arturmusin. This discussions form is for suggesting features for the npm registry, website, and CLI. We do not offer general technical support through this channel. You may want to try opening an issue over at https://github.community or on Stackoverflow. Best of luck!
